src/main/java/datawave/query/util/MetadataHelper.java

```diff
@@ -322,16 +322,12 @@ public Set<String> getAllFields(Set<String> ingestTypeFilter) throws TableNotFou
             log.trace("loadAllFields() with auths:" + this.allFieldMetadataHelper.getAuths() + " returned " + allFields);
         
         Set<String> fields = new HashSet<>();
-        if (ingestTypeFilter == null || ingestTypeFilter.isEmpty()) {
-            fields.addAll(allFields.values());
-        } else {
-            for (String datatype : ingestTypeFilter) {
-                fields.addAll(allFields.get(datatype));
-            }
-        }
+        fields.addAll(getFields(allFields, ingestTypeFilter));
         
         // Add any additional fields that are created at evaluation time and are hence not in the metadata table.
-        fields.addAll(evaluationOnlyFields);
+        if (!evaluationOnlyFields.isEmpty()) {
```

Comment:
no need to check for empty here, but ok.

Reply:
+1